test/conformance/ingress/rewrite.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ingress
+
+import (
+	"testing"
+
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"knative.dev/networking/pkg/apis/networking"
+	"knative.dev/networking/pkg/apis/networking/v1alpha1"
+	"knative.dev/serving/test"
+)
+
+// TestRewriteHost verifies that a RewriteHost rule can be used to implement vanity URLs.
+func TestRewriteHost(t *testing.T) {
+	t.Parallel()
+	clients := test.Setup(t)
+
+	name, port, cancel := CreateRuntimeService(t, clients, networking.ServicePortNameHTTP1)
+	defer cancel()
+
+	// Create a simple Ingress over the Service.
+	_, _, cancel = CreateIngressReady(t, clients, v1alpha1.IngressSpec{
+		Rules: []v1alpha1.IngressRule{{
+			Visibility: v1alpha1.IngressVisibilityClusterLocal,
+			Hosts:      []string{name + ".example.com"},
+			HTTP: &v1alpha1.HTTPIngressRuleValue{
+				Paths: []v1alpha1.HTTPIngressPath{{
+					Splits: []v1alpha1.IngressBackendSplit{{
+						IngressBackend: v1alpha1.IngressBackend{
+							ServiceName:      name,
+							ServiceNamespace: test.ServingNamespace,
+							ServicePort:      intstr.FromInt(port),
+						},
+					}},
+				}},
+			},
+		}},
+	})
+	defer cancel()
+
+	// Now create a RewriteHost ingress to point a custom Host at the Service
+	_, client, cancel := CreateIngressReady(t, clients, v1alpha1.IngressSpec{
+		Rules: []v1alpha1.IngressRule{{
+			Hosts:      []string{"vanity.ismy.name", "vanity.isalsomy.number"},
+			Visibility: v1alpha1.IngressVisibilityExternalIP,
+			HTTP: &v1alpha1.HTTPIngressRuleValue{
+				Paths: []v1alpha1.HTTPIngressPath{{
+					RewriteHost: name + ".example.com",
+				}},
+			},
+		}},
+	})
+	defer cancel()
+
+	RuntimeRequest(t, client, "http://vanity.ismy.name")
+	RuntimeRequest(t, client, "http://vanity.isalsomy.number")
```

Comment:
these assertions are pretty weak - do we want to assert anything beyond the HTTP 200?

Reply:
this seems consistent with all the other tests in conformance/ingress from what I could see, but I'm not super familiar with them tbh.. is there an example of another ingress conformance test that does more?

Reply:
@tcnghia do you have any guidance?

Reply:
Yes, `RuntimeRequest` is the general assertion that we have. There are also `RuntimeRequestWithExpectation` that adds additional testing, but I think this is a good starting point.